src/server/pps/server/api_server.go

```diff
@@ -2166,14 +2201,13 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *pps.CreatePipel
 
 // setPipelineDefaults sets the default values for a pipeline info
 func setPipelineDefaults(pipelineInfo *pps.PipelineInfo) error {
-	if pipelineInfo.Transform == nil {
-		return fmt.Errorf("pipeline spec is missing transform: %v", pipelineInfo)
-	}
-
 	now := time.Now()
 	if pipelineInfo.Transform.Image == "" {
 		pipelineInfo.Transform.Image = DefaultUserImage
 	}
+	if pipelineInfo.Salt == "" {
+		pipelineInfo.Salt = uuid.NewWithoutDashes()
+	}
```

Comment:
I'm not sure if it's good to move this here. I just made some changes here: #4232 that this is going to conflict with. And I think because of those changes we need to leave the salt generation in the same place because we want to have access to the `request` when we decide whether or not to fill it in so we can tell if reprocess was called.

Reply:
:+1: moved it back to `CreatePipeline()` where it was before